red/runtime/storage/localfilesystem/projects/Project.js

```diff
@@ -701,6 +703,15 @@ function createProject(user, metadata) {
                         );
                         auth = authCache.get(project,originRemote.url,username);
                     }
+                    else if (originRemote.hasOwnProperty("key_file") && originRemote.hasOwnProperty("passphrase")) {
+                        var key_file_name = (username === '_') ? os.hostname() + '_' + originRemote.key_file : username + '_' + originRemote.key_file;
```

Comment:
See my other comment about using hostname ... think it should be a hardcoded default value instead.